src/state/registration/hooks.ts

```diff
@@ -0,0 +1,10 @@
+import { useWeb3React } from '@web3-react/core'
+import { useAppSelector } from 'state/hooks'
+
+export function useIsUserRegisteredWithViolet(): boolean | undefined {
+  const { account } = useWeb3React()
```

Comment:
Do you think it would be better to pass the wallet on the action payload instead of instantiating it here? I'm afraid this way we could get into some inconsistencies on edge cases

Reply:
Ah you're right, let's do that :D

Reply:
Fixed in https://github.com/violetprotocol/mauve-dex/pull/153/commits/c75fd8c19419a8980872544d0048a09ac7b1f14e